core/src/main/java/org/mskcc/cbio/portal/dao/DaoCnaEvent.java

```diff
@@ -147,38 +159,53 @@ private static long addCnaEventDirectly(CnaEvent cnaEvent) throws DaoException {
         try {
             con = JdbcUtil.getDbConnection(DaoCnaEvent.class);
             pstmt = con.prepareStatement
-		("SELECT sample_cna_event.CNA_EVENT_ID, SAMPLE_ID, GENETIC_PROFILE_ID,"
-                    + " ENTREZ_GENE_ID, ALTERATION FROM sample_cna_event, cna_event"
-                    + " WHERE `GENETIC_PROFILE_ID`=?"
+		("SELECT sample_cna_event.CNA_EVENT_ID,"
+                    + " sample_cna_event.SAMPLE_ID,"
+                    + " sample_cna_event.GENETIC_PROFILE_ID,"
+                    + " ENTREZ_GENE_ID,"
+                    + " ALTERATION,"
+                    + " alteration_driver_annotation.DRIVER_FILTER,"
+                    + " alteration_driver_annotation.DRIVER_FILTER_ANNOTATION,"
+                    + " alteration_driver_annotation.DRIVER_TIERS_FILTER,"
+                    + " alteration_driver_annotation.DRIVER_TIERS_FILTER_ANNOTATION"
+                    + " FROM sample_cna_event"
+                    + " LEFT JOIN alteration_driver_annotation ON"
```

Comment:
has this query been tested on a db with large number of records/large study? Just wondering if there was any changes in performance that needs to be considered for periodic imports.

Reply:
No, it has not. Is there a particular way to test this beforehand that you can recommend?

Reply:
btw i did notice this stackoverflow issue re doing a left join with empty rows: https://stackoverflow.com/questions/31207935/mysql-with-left-join-is-slow-with-empty-row. Didn't read carefully to check if this applies here but just sharing for reference. I guess in the public cBioPortal `alteration_driver_annotation` will be empty at first

Could also try with e.g. the latest dump of the public portal: http://download.cbioportal.org/mysql-snapshots/mysql-snapshots-toc.html